api/src/main/java/org/pmiops/workbench/google/CloudStorageServiceImpl.java

```diff
@@ -35,6 +39,62 @@ public String readMandrillApiKey() {
     return mandrillKeys.getString("api-key");
   }
 
+  public void copyAllDemoNotebooks(String workspaceBucket)  {
+    Storage storage = StorageOptions.getDefaultInstance().getService();
+    Bucket demoBucket = storage.get(getDemosBucketName());
+    Page<Blob> allBlobs = demoBucket.list();
+    List<Blob> notebookFiles = new ArrayList<>();
+    notebookFiles.addAll(getFileTypeFromPage(allBlobs, ".ipynb"));
+    while (allBlobs.hasNextPage()) {
+      allBlobs = allBlobs.getNextPage();
+      notebookFiles.addAll(getFileTypeFromPage(allBlobs, ".ipynb"));
+    }
+    notebookFiles.forEach((Blob blob) -> {
+      BlobId demoNotebookId = blob.getBlobId();
+      BlobId targetLocation = BlobId.of(workspaceBucket, "notebooks/" + demoNotebookId.getName());
+      copyBlob(demoNotebookId, targetLocation);
+    });
+  }
+
+  private List<Blob> getFileTypeFromPage(Page<Blob> page, String extension) {
+    List<Blob> fileList = new ArrayList<>();
+    Iterable<Blob> iter = page.iterateAll();
+    iter.forEach((Blob blob) -> {
+      if (blob.getBlobId().getName().endsWith(extension)) {
+        fileList.add(blob);
+      }
+    });
+    return fileList;
+  }
+
+  private List<JSONObject> getJsonAndFilterByTypeFromPage(Page<Blob> page, String type) {
```

Comment:
`type` is always cohort here. My preference for comparisons like this is to use enums instead of raw strings. But, I'm not sure if this type is initially set on the cohort object, or if this is the same thing as the `type` field on the cohort json object (I'm guessing the latter?). If it's also set in code somewhere else, that means more refactoring, so I'm open to letting this go.

Reply:
This is actually something that Fritz wanted when I talked to him so that we could add more json demos that are different things later (future proofing)